### PR TITLE
Agent Health Endpoint

### PR DESCRIPTION
**Pull request by schmichael**

### CODE REVIEW

File: command/agent/agent_endpoint.go [hunk 1]
```diff
@@ -299,3 +299,92 @@ type joinResult struct {
...
+			health.Server.Message = err.Error()
+		} else if leader == "" {
+			health.Server.Ok = false
+			health.Server.Message = "no leader"
+		}
+	}
+
+	if !health.ok() {
+		// At least one not-ok response, set failing status code
+		resp.WriteHeader(500)
```
Comment (dadgar) [inline command/agent/agent_endpoint.go]: Use `CodedError`: https://github.com/hashicorp/nomad/blob/4fb2dce22a0c82b969a1715c0f1ac7099cd5e331/command/agent/deployment_endpoint.go#L12

File: command/agent/agent_endpoint.go [hunk 2]
```diff
@@ -299,3 +299,92 @@ type joinResult struct {
...
+					Ok:      false,
+					Message: "server not enabled",
+				}
+			}
+		}
+	}
+
+	// If we should check the client and it exists assume it's healthy
+	if client := s.agent.Client(); getClient && client != nil {
+		health.Client = &healthResponseAgent{
```
Comment (dadgar) [inline command/agent/agent_endpoint.go]: What about:
```
if len(client.GetServers()) == 0 {
  health.Client = &healthResponseAgent{
    Ok: false,
    Message: "no known servers",
  }
} else {
  health.Client = &healthResponseAgent{
    Ok: true,
    Message: "ok",
  }
}

File: command/agent/agent_endpoint.go (hunk 1 above)
Comment (schmichael) [reply command/agent/agent_endpoint.go]: To use that and keep the JSON format I'd have to serialize the `healthResponse` struct to a string and use that for an error. This just seemed simpler.

Comment (schmichael) [reply command/agent/agent_endpoint.go]: Oh but this breaks the Content Type header... fixing

**Pull request merged**